### PR TITLE
Updates for 1.36

### PR DESCRIPTION
**Pull request by Chuxel**
This PR includes the a set of Remote Development edits for the 1.36 release. (Right now it shows tons of changes since vNext does not the latest changes from master in it, but that will change once #2823 is merged.)

Specifically, it:

1. [SSH/Containers] Adds new information on ARM32 and Alpine support. It:
    1. Calls out support in a new "System Requirements" part in each doc along with needed clarifications. 
    2. Adds these same details to the Linux Prerequisites article.
    3. Explains why certain extensions may not work for these platforms in all three places.
    4. Adds information in tips and tricks and the extension guide about these extension issues.
5. [SSH] Tunes some of the SSH guidance now that we prompt for passwords but highlights that you will still have to enter information twice.
6. [SSH] Adds a pointer to tips and tricks content on SSHFS and rsync setup in the primary SSH article (to elevate visibility a bit given the number of people asking about mounting / syncing the remote).
6. [Containers] Drops references to mounting `~/.gitignore` now that this is copied, but adds information on mounting `~/.ssh` in a new section given how frequently git SSH key management has come up.
7. [Containers] Mentions postCreateCommand as a way to install dev dependencies without altering an existing Dockerfile or image.
8. [Containers] Raises the advanced configuration sub-section in the main article that points to the Advanced Configuration article to an H2 since some customers who have read the main docs have missed it. There's also some related language simplification in the advanced config article.
8. Misc clarifications and cleanup like updating command names.

//cc: @roblourens @joaomoreno @chrmarti

### CODE REVIEW

File: docs/remote/containers-advanced.md [hunk 1]
```diff
@@ -421,17 +447,15 @@ Follow these steps:
...
 
     ```json
-    {
-      "docker.host":"tcp://localhost:23750"
-    }
+    "docker.host":"tcp://localhost:23750"
     ```
 
-5. Next, use the file mount to copy `~/repos/your-repository-name/.devcontainer/devcontainer.json` to `.devcontainer.json` (dot-prefixed) in `~/devcontainers/repository-name-here` and make a few edits. In this example, we are using a Dockerfile, so the changes would be as follows:
+5. Next, use the file mount to copy `~/repos/your-repository-name/.devcontainer/devcontainer.json` to `.~~devcontainer~~.json` (dot-prefixed) in `~/devcontainers/repository-name-here` and update the default workspace mount in the file to point to the source code on the remote machine. In this example, we are using a Dockerfile, so the changes would be as follows:
```
Comment (chrmarti) [inline docs/remote/containers-advanced.md]: The double `~~` look out of place, are these Markdown or typos?

File: docs/remote/containers.md [hunk 2]
```diff
@@ -19,13 +19,25 @@ This lets VS Code provide a **local-quality development experience** — includi
 
 ## Getting started
 
+### System Requirements
+
+**Local:** Docker Desktop 2.0+ for macOS/Windows or Docker CE/EE 18.06+ and Docker Compose 1.21+ for Linux. Docker Toolbox and Ubuntu snap packages are not supported. For Windows, Windows 10 Professional or Enterprise is required due to the requirements of Docker Desktop or Windows. See the minimum requirements for [VS Code](/docs/supporting/requirements.md) for additional details.
```
Comment (chrmarti) [inline docs/remote/containers.md]: Compose 1.21 is only needed when the project folder name contains hyphens or underscores.

File: docs/remote/containers.md [hunk 3]
```diff
@@ -568,11 +646,7 @@ VS Code will then **automatically use both files** when starting up any containe
 docker-compose -f docker-compose.yml -f .devcontainer/docker-compose.extend.yml up
 ```
 
-### Using an updated Dockerfile to automatically install more tools
-
-You may want to install other tools such as Git inside the container for the service you've specified. You can easily [do this manually](#installing-additional-software-in-the-sandbox). However, you can also create a custom `Dockerfile` specifically for development that includes these dependencies. The [vscode-dev-containers repository](https://github.com/Microsoft/vscode-dev-containers) contains examples you can use to augment a copy of a Dockerfile or when creating a new one.
-
-Assuming you put this file under `.devcontainer/Dockerfile`, the `.devcontainer/docker-compose.yml` above would be modified as follows:
+While the `postCreateCommand` property allows you to install additional tools inside your container, in some cases you may want to have a specific Dockerfile for development. You can also use this same approach to reference a custom `Dockerfile` specifically for development without modifying your existing Docker Compose file.  For example, you can update `.devcontainer/devcontainer.extend.json` as follows:
```
Comment (chrmarti) [inline docs/remote/containers.md]: Should the filename be `docker-compose.extend.yml`?

File: api/advanced-topics/remote-extensions.md [hunk 4]
```diff
@@ -355,9 +355,28 @@ To solve this problem:
...
+
+For example, your extension may only include x86_64 compiled versions of native modules or runtimes. For Alpine Linux, the included native code or runtimes may not work due to [fundamental differences](https://wiki.musl-libc.org/functional-differences-from-glibc.html) between how `libc` is implemented in Alpine Linux (`musl`) and other distributions (`glibc`).
+
+To resolve this problem:
+
+1. If you are dynamically acquiring compiled code, you can add support by detecting non-x86_64 targets using `process.arch` and downloading versions compiled for the right architecture. If you are including binaries for all supported architectures inside your extension instead, you can use this logic to use the correct one.
+
+2. For Alpine Linux, you can use `fs.existsSync('/etc/alpine-release')` and once again download or use the correct binaries for a `muscl` based operating system.
```
Comment (roblourens) [inline api/advanced-topics/remote-extensions.md]: The example should use the async version, we shouldn't encourage extensions to do sync IO

File: docs/remote/containers.md (hunk 2 above)
Comment (Chuxel) [reply docs/remote/containers.md]: Yeah I thought it would be easier just to recommend 1.21 -- and all of our "try" and dev contaienr defintion test projects end up having a dash in them because of the folder name.  We're also all on latest, so there's potentially other issues we just don't know about yet.

File: docs/remote/containers.md (hunk 3 above)
Comment (Chuxel) [reply docs/remote/containers.md]: Good catch - that's actually been there a while and I totally missed it on re-reads.

File: docs/remote/ssh.md [hunk 5]
```diff
@@ -19,33 +19,45 @@ This lets VS Code provide a **local-quality development experience** — includi
...
     > **Note:** PuTTY is not supported on Windows since the `ssh` command must be in the path.
 
-3. Install [Visual Studio Code](https://code.visualstudio.com/).
+2. Install [Visual Studio Code](https://code.visualstudio.com/) or [Visual Studio Code Insiders](https://code.visualstudio.com/insiders/).
 
-4. Install the [Remote Development](https://aka.ms/vscode-remote/download/extension) extension pack.
+3. Install the [Remote Development](https://aka.ms/vscode-remote/download/extension) extension pack.
 
-5. [Optional] If your server requires multi-factor authentication, set `"remote.SSH.showLoginTerminal": true` in `settings.json` and enable the `ControlMaster` SSH feature. [See here for details](/docs/remote/troubleshooting.md#enabling-alternate-ssh-authentication-methods).
+4. [Optional] If you need to enter a token or password when connecting to your host, [you can enable `ControlMaster` in your SSH config](/docs/remote/troubleshooting.md#enabling-alternate-ssh-authentication-methods) to prevent you from having to enter it multiple times. However, we have seen mixed results with this setting and Windows SSH clients.
```
Comment (roblourens) [inline docs/remote/ssh.md]: Does it work with any windows client? I think it's not supported on windows at all.

File: docs/remote/containers-advanced.md (hunk 1 above)
Comment (Chuxel) [reply docs/remote/containers-advanced.md]: Thanks - Yeah "ctrl+s" on mac does that with markdown all-in-one extension installed.  Must have been jumping between my windows and mac at the time.

File: docs/remote/troubleshooting.md [hunk 6]
```diff
@@ -185,9 +185,9 @@ If you are connecting to an SSH remote host and are either:
 - using password authentication,
 - using an SSH key with a passphrase when the [SSH Agent](#setting-up-the-ssh-agent) is not running or accessible,
 
-...you need to enable the `remote.SSH.showLoginTerminal` [setting](/docs/getstarted/settings.md) in VS Code. This setting displays the terminal whenever VS Code runs an SSH command. You can then enter your auth code, password, or passphrase when the terminal appears.
+...VS Code should automatically prompt you to enter needed information. If you do not see the prompt, enable the `remote.SSH.showLoginTerminal` [setting](/docs/getsarted/settings.md) in VS Code. This setting displays the terminal whenever VS Code runs an SSH command. You can then enter your auth code, password, or passphrase when the terminal appears.
```
Comment (roblourens) [inline docs/remote/troubleshooting.md]: "getsarted" was that change intentional?

File: docs/remote/ssh.md (hunk 5 above)
Comment (Chuxel) [reply docs/remote/ssh.md]: Yeah I wasn't 100% sure, so I didn't lock in, but if you haven't heard of one either let's just say it's not supported.

File: api/advanced-topics/remote-extensions.md (hunk 4 above)
Comment (Chuxel) [reply api/advanced-topics/remote-extensions.md]: Good point.  Can do that.

File: docs/remote/troubleshooting.md (hunk 6 above)
Comment (Chuxel) [reply docs/remote/troubleshooting.md]: Nope!
